app.py

```diff
@@ -0,0 +1,233 @@
+            wants_accommodation = map_[arg['--wants_accomm'].upper()]
+        except KeyError:
+            print("Invalid option: " + arg['--wants_accomm'])
+            return
+
+        status = self.amity.add_person(person_name, person_type,
+                                       wants_accommodation=wants_accommodation)
+
+        if type(status) != str:
+            cprint('Person added successfully!', 'green')
+
+        else:
+            cprint(status, 'cyan')
+
+    @docopt_cmd
+    def do_reallocate_person(self, arg):
+        '''
+        Usage: reallocate_person <first_name> <second_name> <room_name>
+
+        Move person from the current room to a specified room
+        '''
+
+        room_name = arg['<room_name>']
+        person_name = arg['<first_name>'] + ' ' + arg['<second_name>']
+
+        status = self.amity.reallocate_person(person_name, room_name)
+        cprint(status, 'cyan')
+
+    @docopt_cmd
+    def do_load_people(self, arg):
+        '''
+        Usage: load_people <filepath>
+
+        Loads people into amity from file
+        '''
+        filepath = arg["<filepath>"]
+        status = self.amity.load_people(filepath)
+        cprint(status, 'cyan')
+
+    @docopt_cmd
+    def do_print_allocations(self, arg):
+        '''
+        Usage: print_allocations [--o=filename]
+
+        Prints out each room's name and the people in it;
+        this can be piped into a file if specified in the filename parameter.
+        Each file is saved in the directory textfiles.
+        '''
+        filename = arg['--o']
+
+        if filename != None:
+            status = self.amity.print_allocations(filename=filename)
+        else:
+            status = self.amity.print_allocations()
+        cprint(status, 'cyan')
+
+    @docopt_cmd
+    def do_print_unallocated(self, arg):
+        '''
+        Usage: print_unallocated [--o=filename]
+
+        Print list of people not in a room
+        '''
+        filename = arg['--o']
+
+        if filename != None:
+            status = self.amity.print_unallocated(filename=filename)
+        else:
+            status = self.amity.print_unallocated()
+
+        cprint(status, 'cyan')
+
+    @docopt_cmd
+    def do_print_room(self, arg):
+        '''
+        Usage: print_room <room_name>
+
+        Print occupants of a room
+        '''
+
+        room_name = arg['<room_name>']
+        status = self.amity.print_room(room_name)
+        if not type(status) is str:
+            for name in status:
+                print(name)
+            return
+        cprint(status, 'cyan')
+
+    @docopt_cmd
+    def do_save_state(self, arg):
+        '''
+        Usage: save_state [--db=sqlite_db]
+
+        Persists the data of the app in database and stores it in the
+        /databases folder
+        '''
+        db = arg['--db']
+
+        if db is not None:
+            status = self.amity.save_state(db=db)
+        else:
+            status = self.amity.save_state()
```

Comment:
if the `save_state` has a default db then the comment on L192 may not be valid, neither might this if statement..L195 alone would be sufficient